src/plugins/engagement/public/components/iframe_chat/iframe_chat.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useRef, useState, CSSProperties } from 'react';
+import { css } from '@emotion/react';
+import { useChat } from '../../services';
+
+type UseChatType =
+  | [false]
+  | [true, string, React.MutableRefObject<HTMLIFrameElement | null>, CSSProperties];
```

Comment:
NIT: a structure would probably be more explicit for this polymorphic type?


```ts
type UseChatType = 
   { enabled: false } 
 | { 
      enabled: true, 
      url: string, 
      elementRef: React.MutableRefObject<HTMLIFrameElement | null>, 
      cssProps: CSSProperties
    }
```

Reply:
Nice.